Only running karma test, dojo-webpack-plugin@3.0.1 DojoAMDMainTemplate.runtime.js::resolveTernaryHasExpression() throws "Module not found: dojo/has" error

When running karma/jasmine tests (actual app works fine -- only tests fail),

dojo-webpack-plugin@3.0.1 DojoAMDMainTemplate.runtime.js::resolveTernaryHasExpression() throws "Module not found: dojo/has" error 
for any source file that requires "dojo/on", or anything that includes "dojo/on", which does this: 
on.js:
  define(["./has!dom-addeventlistener?:./aspect", "./_base/kernel", "./sniff"], function(aspect, dojo, has)
The reason seems to be that req.absMids is empty inside findModule when it is called by resolveTernaryHasExpression.

dojo-webpack-plugin@2.8.19  works fine with the same sources.

dojo-webpack-plugin 2.8.19

runtime.js

 
        (this || window)["webpackJsonp"].registerAbsMids = registerAbsMids;
 

commons.js

    if (jsonpArray.registerAbsMids) {
        jsonpArray.registerAbsMids(absMids);   // IT DOES THIS
    } else {
        var absMidsWaiting = jsonpArray.absMidsWaiting = jsonpArray.absMidsWaiting || [];
        absMidsWaiting.push(absMids);
    }
}

function() {
    var absMids = {
        "cloudCenterRouter": "./assets/js/cloudCenterRouter.js",
        "dao/abstractDao": "./assets/js/dao/abstractDao.js",
        "dao/mockDao": "./assets/js/dao/mockDao.js",
        "config": "./config/default-dev/config.js",
        // "/Users/gconnell/github/cloudCenter/web-client-application/node_modules/dojo-webpack-plugin/lib/NoModule.js" = "./node_modules/dojo-webpack-plugin/lib/NoModule.js"
        // "./../../../../../../node_modules/webpack/buildin/global.js" = "./node_modules/webpack/buildin/global.js"
        "backbone": "./target/debug/2P/derived/ui/Backbone/backbone.js",
        "underscore": "./target/debug/2P/derived/ui/Underscore/underscore.min.js",
        "dojo/_base/config": "./target/debug/2P/derived/ui/dojo/_base/config.js",
        "dojo/_base/kernel": "./target/debug/2P/derived/ui/dojo/_base/kernel.js",
        "dojo/aspect": "./target/debug/2P/derived/ui/dojo/aspect.js",
        "dojo/has": "./target/debug/2P/derived/ui/dojo/has.js",
        "dojo/on": "./target/debug/2P/derived/ui/dojo/on.js",
        "dojo/sniff": "./target/debug/2P/derived/ui/dojo/sniff.js",
        "jquery": "./target/debug/2P/derived/ui/jQuery/jquery-2.2.0.min.js"// "/Users/gconnell/github/cloudCenter/web-client-application/test/spec/CloudCenterRouterSpec.js" = "./test/spec/CloudCenterRouterSpec.js"
    };
    var globalObj = this || window;
    var jsonpArray = globalObj["webpackJsonp"] = globalObj["webpackJsonp"] || [];
    if (jsonpArray.registerAbsMids) {
        jsonpArray.registerAbsMids(absMids);  // IT DOES THIS
    } else {
        var absMidsWaiting = jsonpArray.absMidsWaiting = jsonpArray.absMidsWaiting || [];
        absMidsWaiting.push(absMids);
    }
}
)()

 resolveTernaryHasExpression(expr)  called with expr = "dom-addeventlistener?:./target/debug/2P/derived/ui/dojo/aspect.js"

findModule(mid, referenceModule, noInstall, asModuleObj) call with mid = "dojo/has", referenceModule = null, noInstall = false, asModuleObj = undefined.

Inside findModule, 
if (mid in req.absMids && __webpack_require__.m[req.absMids[mid]])  is TRUE because req.absMids = {
backbone: "./target/debug/2P/derived/ui/Backbone/backbone.js"
cloudCenterRouter: "./assets/js/cloudCenterRouter.js"
config: "./config/default-dev/config.js"
dao/abstractDao: "./assets/js/dao/abstractDao.js"
dao/mockDao: "./assets/js/dao/mockDao.js"
dojo/_base/config: "./target/debug/2P/derived/ui/dojo/_base/config.js"
dojo/_base/kernel: "./target/debug/2P/derived/ui/dojo/_base/kernel.js"
dojo/aspect: "./target/debug/2P/derived/ui/dojo/aspect.js"
dojo/has: "./target/debug/2P/derived/ui/dojo/has.js"
dojo/on: "./target/debug/2P/derived/ui/dojo/on.js"
dojo/sniff: "./target/debug/2P/derived/ui/dojo/sniff.js"
jquery: "./target/debug/2P/derived/ui/jQuery/jquery-2.2.0.min.js"
underscore: "./target/debug/2P/derived/ui/Underscore/underscore.min.js"
__proto__: Object
}

It finds "dojo/has" and all works.

dojo-webpack-plugin 3.0.1

commons.js

(function() {
    var absMids = {
        "cloudCenterRouter": 4,
        "dao/abstractDao": 14,
        "dao/mockDao": 13,
        "config": 15,
        // "C:\\Users\\Glen\\github\\cloudcenter\\web-client-application\\node_modules\\dojo-webpack-plugin\\lib\\NoModule.js" = 8
        "backbone": 5,
        "underscore": 1,
        "dojo/_base/config": 10,
        "dojo/_base/kernel": 9,
        "dojo/aspect": 12,
        "dojo/has": 6,
        "dojo/on": 7,
        "dojo/sniff": 11,
        "jquery": 2 // "C:/Users/Glen/github/cloudcenter/web-client-application/test/spec/CloudCenterRouterSpec.js" = 0
    };
    var globalObj = this || window;
    var jsonpArray = globalObj["webpackChunkcloudcenter2_webapp"] = globalObj["webpackChunkcloudcenter2_webapp"] || [];
    if (jsonpArray.registerAbsMids) {
        jsonpArray.registerAbsMids(absMids);
    } else {
        var absMidsWaiting = jsonpArray.absMidsWaiting = jsonpArray.absMidsWaiting || []; // IT DOES THIS because registerAbsMids never assigned to jsonpArray
        absMidsWaiting.push(absMids);
    }
} 
 
runtime.js

resolveTernaryHasExpression(expr)  called with expr = "dom-addeventlistener?:12"

findModule(mid, referenceModule, noInstall, asModuleObj) call with mid = "dojo/has", referenceModule = null, noInstall = false, asModuleObj = undefined.
Inside fineModule, 
if (mid in req.absMids && __webpack_require__.m[req.absMids[mid]])  is FALSE because req.absMids = {__proto__: Object}  It's empty so, get error "Module not found: dojo/has".

When I run `npm install` I get the following error:
```
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/karma-mw-browser-support - Not found
npm ERR! 404
npm ERR! 404  'karma-mw-browser-support@0.0.1-pre.2021.2.24.19.15.3' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 It was specified as a dependency of 'bugrepro'
```
It seems that you're loading this from a local artifactory repo

--- a/lib/DojoAMDRuntimeModule.js
+++ b/lib/DojoAMDRuntimeModule.js
@@ -40,11 +40,9 @@ class DojoAMDRuntimeModule extends RuntimeModule {
 		buf.push("});");
 		buf.push("");
 		buf.push(pluginProps.hooks.dojoGlobalRequire.call("globalObj.require = req;"));
-		if(this.chunk.hasAsyncChunks()) {
-			buf.push(Template.indent(
-				`(self[${chunkLoadingGlobal}] = self[${chunkLoadingGlobal}] || []).registerAbsMids = registerAbsMids;`
-			));
-		}
+		buf.push(Template.indent(
+			`(self[${chunkLoadingGlobal}] = self[${chunkLoadingGlobal}] || []).registerAbsMids = registerAbsMids;`
+		));
 		buf.push("");
 		buf.push("// expose the Dojo compatibility functions as a properties of __webpack_require__");
 		buf.push(`if (${djProp} && ${djProp}.name !== '${pluginName}') throw new Error("${djProp} name collision.")`);
@@ -104,13 +102,11 @@ but the loader specified at ${pluginProps.embeddedLoaderFilename} was built with
 		if (loaderConfig.has && loaderConfig.has['dojo-undef-api']) {
 			buf.push("req.undef = " +  Template.getFunctionContent(runtime.undef));
 		}
-		if(this.chunk.hasAsyncChunks()) {
-			buf.push(`var absMidsWaiting = globalObj[${chunkLoadingGlobal}].absMidsWaiting;`);
-			buf.push("if (absMidsWaiting) {");
-			buf.push("   absMidsWaiting.forEach(registerAbsMids);");
-			buf.push(`   delete globalObj[${chunkLoadingGlobal}].absMidsWaiting;`);
-			buf.push("}");
-		}
+		buf.push(`var absMidsWaiting = globalObj[${chunkLoadingGlobal}].absMidsWaiting;`);
+		buf.push("if (absMidsWaiting) {");
+		buf.push("   absMidsWaiting.forEach(registerAbsMids);");
+		buf.push(`   delete globalObj[${chunkLoadingGlobal}].absMidsWaiting;`);
+		buf.push("}");
 		return Template.asString([
 			`(function() { /* Start ${pluginName} extensions */`,
 			Template.indent(buf),

